src/authorizations/utils/permissions.ts

```diff
@@ -135,3 +135,72 @@ export enum BucketTab {
   AllBuckets = 'All Buckets',
   Scoped = 'Scoped',
 }
+
+export const formatPermissionsObj = permissions => {
+  const newPerms = permissions.reduce((acc, {action, resource}) => {
+    const {type, id, orgID, name} = resource
+    let p
```

Comment:
```ts
let p
```
<img width="80%" height="80%" src="https://user-images.githubusercontent.com/146112/133485919-3ec8c828-dd3d-440b-bf8d-2837a85999b9.gif" />
😆😆😆😆

giffing aside, I'm not a very big fan of single letter variable names because they greatly increase the ambiguity of the code. Now it's pretty obvious that in the context of this code review `p` represents permissions, but the problem is that we're looking at this in the context of a code review about permissions. Looking at this code in isolation during a normal operation, it may not be clear what `p` is. Going from the other direction, if we wanted to search the entire codebase for `permission` we may miss this variable, or its usage in the context of a page of search results may not be very clear.

Also, we could remove an `else` case many lines away from the variable being created, and increase the understanding of the code if we did something like

```ts
let permissions = {
  read: action === 'read',
  write: action === 'write',
}

Reply:
you know what, i deserved that meme LOL. we even went back to refactor the code but that one flew over our head. yup! `let p` is really dumb